run.py

```diff
@@ -15,6 +25,49 @@
 })
 
 
+# @event.listens_for(User.password, 'set', retval=True)
+# def hash_user_password(target, value, oldvalue, initiator):
+#     """Encrypts password when new admin created in User View"""
+#     if value != oldvalue:
+#         return utils.encrypt_password(value)
+#     return value
+
+
+@security.context_processor
+def security_context_processor():
+    return dict(
+        admin_base_template=admin.base_template,
+        admin_view=admin.index_view,
+        h=admin_helpers,
+        get_url=url_for
+    )
+
+
 @app.shell_context_processor
 def make_shell_context():
-    return {'db': db, 'Resource': Resource, 'Category': Category, 'Language': Language}
+    return {'db': db, 'Resource': Resource, 'Category': Category, 'Language': Language,
+            'User': User, 'Role': Role}
+
+
+@app.before_first_request
+def before_first_request():
```

Comment:
If the admin user already exists when this function runs, what happens? Are there side effects?